Classes/common/CDTDatastore.m

```diff
@@ -396,6 +396,25 @@ - (BOOL)validateBodyDictionary:(NSDictionary *)body error:(NSError *__autoreleas
             return NO;
         }
     }
+    
+    //now to check if the body contains valid json data
+    if(![NSJSONSerialization isValidJSONObject:body]){
```

Comment:
I'm not sure, but I think the order is correct, because we are checking for keys which are invalid, before checking if the whole thing can be serialised